doc/authoring_help.md

```diff
@@ -0,0 +1,95 @@
+                  type: string
+                  short-summary: 'Account name'
+                  long-summary: |
+                      Longer summary with newlines preserved.  Preserving newlines is helpful for paragraph breaks.
+                  populator-commands: 
+                  - az account list
+                  - These indicate where values can be retrieved for input to this command
+                - name: --another-parameter
+                  short-summary: These parameter names must match what is shown in the command's CLI help output, including abbreviation.
+            examples:
+                - name: Document a parameter that doesn't exist
+                  text: >
+                    You will get an error when you show help for the command stating there is an extra parameter.
+                - name: Collapse whitespace in YAML
+                  text: >
+                    The > character collapses multiple lines into a single line, which is good for on-screen wrapping.
+            """
+</pre>
+
+You can also document groups using the same format.
+
+<pre>
+helps['account'] = """
+            type: group
+            short-summary: The account group
+            long-summary: Longer summary of account            
+            examples:
+                - name: Clear an account 
+                  text: Description
+                - name: Choose your current account
+                  text: az account set...
+            """
+</pre>
+
+# Testing Authored Help #
+
+To verify the YAML help is correctly formatted, the command/group's help command must be executed at runtime.  For example, to verify "az account clear", run the command "az account clear -h" and verify the text.  
+
+Runtime is also when help authoring errors will be reported, such as documenting a parameter that doesn't exist.  Errors will only show when the CLI help is executed, so verifying the CLI help is required to ensure your authoring is correct.   
+
+# Other Help Authoring #
+
+Commands without YAML usually still have help text.  Where does it come from?  These sections briefly outline where Az help text comes from.
+
+Authoring note: it is not recommended to use the product code to author command/group help--YAML is the recommended way (see above).  This information is provided for completeness and may be useful for fixing small typos in existing help text.
+
+## Help Layers ##
+
+Command help starts with its raw SDK docstring text, if available.  Non-SDK commands may have their own docstring.  Code can specify values that replace the SDK/docstring contents.  YAML is the final override for help content and is the recommended way for authoring command and group help.  Note that group help can only be authored via YAML.  
+
+Here are the layers of Project Az help, with each layer overriding the layer below it:
+
+| Help Display   |
+|----------------|
+| YAML Authoring |
+| Code Specified |
+| Docstring      |
+| SDK Text       |
```

Comment:
I was thinking here that we may wrap an SDK call in a custom command, in which case a docstring on the wrapper would override the SDK docstring.  I think it's worthwhile to call out the SDK strings as separate from the product docstrings since they can't be changed.

---
In reply to: [73391384](https://github.com/Azure/azure-cli/pull/600#discussion_r73391384) [](ancestors = 73391384)